### DISCUSSION TITLE
Documenting Neurodocker arguments used to build containers

### DISCUSSION THREAD

**Discussion by civier** [neurodesk/neurodesk.github.io#112] [Ideas]
Hi @stebo85 @aswinnarayanan 

As the ones who approve new containers, I would be happy if you force people to document the arguments they provided for neurodocker for the build. Otherwise, it is very difficult to know if the arguments need to be kept when the container is rebuilt, or whether they were only a temporary solution or other.

I give an example:
I've just built a new bidscoin container based on bidstools. The recipe for bidstools (https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/bidstools/build.sh) included the following arguments for the Neurodocker's --miniconda option:
```
 --miniconda use_env=base \
              conda_install='python=3.6 traits' \
              pip_install='bidscoin' \
```
Now, when I worked on the new container, I didn't have a clue if I need to enforce python 3.6 or not (and why?), and I also didn't know what 'traits' is for. After some trial and error, I found out that using python=3.6 broke the build, so in this case it had to be dropped. If things were documented, that could be easier for me.

If you want to put me as the bad cop, you're welcome to add me to the approval workflow, and I'll do the dirty job. No problem ;-)

Best,
Oren

P.S.
Because the neurodocker command is very long and "magnificent" bash does not allow end-line comments in the middle of commands, people will probably put a comment just before the neurocommand command. That is not optimal, so I suggest we teach people to include end-line comments after all using the following trick:
```
 --miniconda use_env=base                               `# COMMENT 1` \
              conda_install='python=3.6 traits'        `# COMMENT 2` \
              pip_install='bidscoin'                              `# COMMENT 3` \
```
If you approve, I will also edit the template/build.sh file with such comments for example, so people won't forget.

**Comment by stebo85** [comment]
It will be very difficult to document the build of a container consistently, because at build time python 3.6 will work and it will only break later due to the fact that the world moved on. 

Still happy for you to add the comments to the template so developers can do this if they can.

I will look for comments when approving new containers. You already have full privileges in the container repository - so you could approve new container submissions and review them as well.

Kind regards
Steffen

**Comment by civier** [comment]
OK. Sounds good.
@stebo85 @aswinnarayanan  Are there other checks/standards that you enforce when approving new containers? I don't mind taking the job, but I need to know what is expected of me.

**Reply by stebo85** [reply]
no - we currently have no written standards.
